src/app/static/src/tests/components/errorReport.test.ts

```diff
@@ -19,6 +19,8 @@ describe("Error report component", () => {
 
     const generateErrorReport = jest.fn();
 
+    const pattern = RegExp("^\\w+([\\.-]?\\w+)*@\\w+([\\.-]?\\w+)*(\\.\\w{2,3})+$")
```

Comment:
This all looks great but can we have some tests that confirm that this regex does what we want? What sort of invalid emails are users adding? Because that might inform the regex, e.g. it might be enough to just check that the value provided has an "@" sign in it rather than worry too much about the exact structure of the email address. Technically there are some valid email address that would fail this regex e.g. "test+user@gmail.com", "test..user@gmail.com". 

Also might be useful to trim whitespace from the email field first because that's a really common typo and often hard for the user to spot what's wrong if they get the invalid feedback for " test.user@gmail.com" or "test.user@gmail.com ".

Reply:
> 

- I don't think these  "test+user@gmail.com", "test..user@gmail.com" are valid email prefixes. The format I adopted conforms with Microsoft email checks as we will be sending via sharePoint. I will write more tests.
- I will trim the email.

Reply:
They are definitely valid email address, (alex+hill@gmail.com and alex..hill@gmail.com both come to me actually, gmail is a bit funny like that!) Admittedly unlikely that anyone would have email addresses that look like that